lib/parser/builders/default.rb

```diff
@@ -1213,6 +1213,7 @@ def begin_keyword(begin_t, body, end_t)
     #
 
     def case_match(case_t, expr, in_bodies, else_t, else_body, end_t)
+      else_body = n(:blank, nil, token_map(else_t)) if else_t && !else_body
       n(:case_match, [ expr, *(in_bodies << else_body)],
         condition_map(case_t, expr, nil, nil, else_t, else_body, end_t))
```

Comment:
we pass `else_t` here to the source map. It seems to be enough to distinguish no-else/empty-else. Do we really need this `blank` node here?

Reply:
My opinion is that the difference should be a part of AST not metadata. Using `else_t` could be enough but looks like a hack to me.

For example, when calling `ruby-parse` you can not see the difference:

```sh
$ ruby-parse --27 -e 'case a; in a; end'
(case-match
  (send nil :a)
  (in-pattern
    (match-var :a) nil nil) nil)

# without blank node it's the same as above
$ ruby-parse --27 -e 'case a; in a; else; end'
(case-match
  (send nil :a)
  (in-pattern
    (match-var :a) nil nil) nil)

# with blank node
$ ruby-parse --27 -e 'case a; in a; else; end'
(case-match
  (send nil :a)
  (in-pattern
    (match-var :a) nil nil)
  (blank))
```

Reply:
Good point. Let's rename it to `empty-else` then?

Reply:
Done!